http_handler.h

```diff
@@ -12,10 +14,20 @@ struct request;
 class handler {
 public:
 
+    handler(std::string base_url) { this->base_url = base_url;};
```

Comment:
A couple things here:

(1) You should pass by const reference so the string is copied once instead of twice
i. There is also move mechanics in C++11 so you don't even do a single copy, but this is new and not used frequently (plus I'm not even 100% sure about how it works, i.e. is it actually more efficient?)

(2) Use initialization lists for this type of thing http://www.cplusplus.com/forum/articles/17820/

`handler(const std::string& root_url) : base_url(root_url) {}`